src/coreclr/jit/assertionprop.cpp

```diff
@@ -3416,19 +3437,33 @@ bool Compiler::optZeroObjAssertionProp(GenTree* tree, ASSERT_VALARG_TP assertion
         return false;
     }
 
-    unsigned       lclNum         = tree->AsLclVar()->GetLclNum();
+    const unsigned lclNum         = tree->AsLclVar()->GetLclNum();
     AssertionIndex assertionIndex = optLocalAssertionIsEqualOrNotEqual(O1K_LCLVAR, lclNum, O2K_ZEROOBJ, 0, assertions);
     if (assertionIndex == NO_ASSERTION_INDEX)
     {
         return false;
     }
 
+    // TODO: create proper simd zero constant
+    //
+    if (varTypeIsSIMD(tree))
+    {
+        return false;
+    }
```

Comment:
We should skip creating `ZeroObj` assertions for `varTypeIsSIMD`. `ZeroObj` is really meant for `TYP_STRUCT` only.